pyopencl/array.py

```diff
@@ -2673,6 +2676,17 @@ def if_positive(criterion, then_, else_, out=None, queue=None):
     contains *then_[i]* if *criterion[i]>0*, else *else_[i]*.
     """
 
+    if all(isinstance(k, SCALAR_CLASSES) for k in [criterion,
+                                                   then_,
+                                                   else_]):
+        result = np.where(criterion, then_, else_)
+
+        if out is not None:
+            out[...] = result
+            return
```

Comment:
The non-scalar code path unconditionally returns `out`.

Reply:
Whoops fixed in f706943.